src/pages/accounts/components/AccountInfoEdit.tsx

```diff
@@ -0,0 +1,87 @@
+import React from "react";
+import { Grid, Typography, TextField, MenuItem } from "@mui/material";
+import { useAppSelector } from "../../../app/common/hooks";
+import { IAccount, rolesEnum } from "../types";
+
+export function AccountEditForm() {
+  const { selectedAccount } = useAppSelector((state) => state.accountManager);
+
+  const [edited, setEdited] = React.useState<IAccount>({
+    role: selectedAccount?.role ?? rolesEnum.TA,
+    username: selectedAccount?.username ?? "",
```

Comment:
I see in a lot of the code using the optional chaining operator (variable?.accessor). Is there a way for us to test which of these could be and could not be undefined?

Reply:
I am unsure what you mean by testing "test which of these could be and could not be undefined". We can talk further in a call.

Reply:
we talked through discord and I'm good to approve (just need to add that comment about the semantics of ??)